functions.lua

```diff
@@ -56,6 +56,278 @@ local function send_message(player, message)
+
+		if minetest.check_player_privs(request.sender, {tp_admin = true}) == false then
+			send_message(request.sender, S("Request Accepted!"))
+		else
+			if tp.enable_immediate_teleport then return end
+
+			send_message(request.sender, S("Request Accepted!"))
+			return
+		end
+	elseif request.direction == "sender" then
+		source = minetest.get_player_by_name(request.sender)
+		target = minetest.get_player_by_name(request.receiver)
+		chatmsg = S("You are teleporting to @1.", request.sender)
+		-- Could happen if either player disconnects (or timeout); if so just abort
+		if not source
+		or not target then
+			send_message(request.receiver, S("@1 is not online right now.", request.sender))
+			return
+		end
+
+		tp.tpr_teleport_player()
+
+		-- Avoid abusing with area requests
+		target_coords = nil
+
+		send_message(request.receiver, chatmsg)
+
+		if minetest.check_player_privs(request.sender, {tp_admin = true}) == false then
+			send_message(request.sender, S("Request Accepted!"))
+		else
+			if tp.enable_immediate_teleport then return end
+
+			send_message(request.sender, S("Request Accepted!"))
+			return
+		end
+	end
+	return request
+end
+
+function tp.deny_request(id, own)
+	local request = tp.clear_request(id)
+	if own then
+		send_message(request.sender, S("You denied your request sent to @1.", request.receiver))
+		send_message(request.receiver, S("@1 denied their request sent to you.", request.sender))
+	else
+		if request.direction == "area" then
+			send_message(request.sender, S("Area request denied."))
+			send_message(request.receiver, S("You denied the request @1 sent you.", request.sender))
+		elseif request.direction == "receiver" then
+			send_message(request.sender, S("Teleport request denied."))
+			send_message(request.receiver, S("You denied the request @1 sent you.", request.sender))
+			spam_prevention[request.receiver] = request.sender
+		elseif request.direction == "sender" then
+			send_message(request.sender, S("Teleport request denied."))
+			send_message(request.receiver, S("You denied the request @1 sent you.", request.sender))
+			spam_prevention[request.receiver] = request.sender
+		end
+	end
+end
+
+function tp.list_requests(playername)
+	local sent_requests = tp.get_requests(playername, "sender")
+	local received_requests = tp.get_requests(playername, "receiver")
+
+	local formspec
+	if sent_requests.count == 0 and received_requests.count == 0 then
+		formspec = ("size[5,2]label[1,0.3;%s:]"):format(S("Teleport Requests"))
+		formspec = formspec..("label[1,1.2;%s]"):format(S("You have no requests."))
+	else
+		local y = 1
+		local request_list_formspec = ""
+		if sent_requests.count ~= 0 then
+			request_list_formspec = request_list_formspec..("label[0.2,%f;%s:]"):format(y, S("Sent by you"))
+			y = y+0.7
+			for request_id, _ in pairs(sent_requests) do
+				if request_id ~= "count" then
+					local request = request_list[request_id]
+					if request.direction == "receiver" then
```

Comment:
I believe area requests should be shown here as well.
Without the area requests, the formspec is kind of bugged out.

Reply:
Will try to take care of this soon 👍

Reply:
Cool, thanks. Let me know if you need any help with it. 🙂